rfcs/0021-glossary-terms.md

```diff
@@ -0,0 +1,123 @@
+
+## Detailed design
+
+[detailed-design]: #detailed-design
+
+The proposal contains the suggested term with it's associated concept it encapsulates, as well as alternative terms and the rationale for  the suggestion.
+
+The terms proposed are below:
+
+### Block Producer
+**Concept:** A consensus node in the network that is selected for a slot, and generates a block.
+
+**Rationale:** This term is used by other protocols, clearly describes the role duties, and is unambiguous.
+
+**Downsides:** EOS is the most popular chain that uses "block producers" and there is a chance of confusion if people think Coda also has only 21 block producers - but this is a tail risk.
+
+**Usage:** A block producer or block producing node stakes coda to participate in consensus.
+
+**Alternatives:**
+- Validator - used by most other protocols, but not suitable for Coda, as every node can be a "verifier" or "validator" of SNARKs.
+- Staker - tied to a specific consensus model, and doesn't explain what the node does for the network. It also has an edit distance of 2 from snarker, which can cause some confusion.
+- Proposer - currently used in codebase, but unused by other protocols, and also sounds less firm / tied to generating blocks (relative to producer).
+- Prover - currently used to describe the action of generating the block SNARK, but if this that job moves to snark workers, then this term is irrelevant.
+- Miner, Baker, etc - not relevant to Coda's consensus mechanism.
+
+### Block
+**Concept:** a set of transactions that represent a state transition.
+
+**Rationale:** This term aligns with the rest of the industry, and it makes it easier to onboard other blockchain users.
+
+**Downsides:** Perhaps it is not as technically accurate as a transition.
+
+**Usage:** In the Coda network, transactions get added to blocks.
+
+**Alternatives:**
+- External transition - maybe this is more precise, but has the heavy lifting of having to explain a new term for a mostly similar concept. Additionally, the word transition is very similar to transaction, causing confusion at times.
+
+### User Transaction
+**Concept:** A transaction issued by a user - currently a payment or a delegation change
+
+**Rationale:** This term clearly describes the concept, and leaves room for further types of user issued transactions.
+
+**Downsides:** This is a subset of "Transactions", so it is a bit redundant.
+
+**Usage:** There are three types of transactions in the Coda network currently - user transactions, fee transfers, and coinbases.
+
+**Alternatives:**
+- User command - this term is confusing, especially in  docs, as it can be conflated with a CLI command that a user issues.
+
+### Snark worker / Snarker
+**Concept:** A Coda node that produces SNARKs to compress transactions (and in the future, blocks) in the network.
```

Comment:
The part here about blocks is not correct. Block producers compress blocks, Snark workers compress transactions.

Reply:
I thought we were considering a change to shift the block compression work away from block producers to snark workers?

Btw - I hadn't intended these "concepts" to be the public facing glossary yet. Was mainly to align on what terms we move forward with. But your suggestions are great, and I see value in locking the definitions in now as well. So will update to not reflect future considerations

Reply:
First I'm hearing of that. And I don't actually see how that would be possible from a protocol perspective either. Either way, from a public facing perspective, perhaps it's better to just keep what work is actually being snarked by the workers abstract? Depends on the level of granularity you want for this definition, really. Though I don't see much harm in a simple high level explanation like this.

Reply:
i'll defer to @bkase - I think that's where I first heard of this potential change (not requiring BPs to generate proofs)?